### ISSUE TITLE
iOS App: crash when try to edit chart in writer document

### ISSUE THREAD

**Issue by nikslor** [CollaboraOnline/online#3293]
**Describe the bug**
Editing a chart in a writer document seems to crash the app in general.

**To Reproduce**
1. Open/create a writer document using the iOS app
2. Tap on "Insert > Chart..."
3. Tap on "Edit Chart" in the sidebar

**Expected behavior**
The dialog to edit the chart should open-

**Actual behavior**
The app crashes

**Tablet**
 - OS: iPadOS 14.6
 - Version: 21.6.1 (1)

**Comment by tml1024** [comment]
Stack trace: `this` is null at
```
#3	0x0000000102ecdaf4 in SvxLineStyleToolBoxControl::setLineStyleIsNoneFunction(std::__1::function<void (bool)> const&) at /Users/tml/lo/co-2021-ios-optimised/svx/source/tbxctrls/linectrl.cxx:182
#4	0x0000000103187e50 in svx::sidebar::LinePropertyPanelBase::Initialize() at /Users/tml/lo/co-2021-ios-optimised/svx/source/sidebar/line/LinePropertyPanelBase.cxx:140
#5	0x0000000103187894 in svx::sidebar::LinePropertyPanelBase::LinePropertyPanelBase(weld::Widget*, com::sun::star::uno::Reference<com::sun::star::frame::XFrame> const&) at /Users/tml/lo/co-2021-ios-optimised/svx/source/sidebar/line/LinePropertyPanelBase.cxx:95
#6	0x000000010068b140 in chart::sidebar::ChartLinePanel::ChartLinePanel(weld::Widget*, com::sun::star::uno::Reference<com::sun::star::frame::XFrame> const&, chart::ChartController*) at /Users/tml/lo/co-2021-ios-optimised/chart2/source/controller/sidebar/ChartLinePanel.cxx:126
#7	0x000000010068aeb4 in chart::sidebar::ChartLinePanel::ChartLinePanel(weld::Widget*, com::sun::star::uno::Reference<com::sun::star::frame::XFrame> const&, chart::ChartController*) [inlined] at /Users/tml/lo/co-2021-ios-optimised/chart2/source/controller/sidebar/ChartLinePanel.cxx:134
#8	0x000000010068aea4 in std::__1::__unique_if<chart::sidebar::ChartLinePanel>::__unique_single std::__1::make_unique<chart::sidebar::ChartLinePanel, weld::Widget*&, com::sun::star::uno::Reference<com::sun::star::frame::XFrame> const&, chart::ChartController*&>(weld::Widget*&, com::sun::star::uno::Reference<com::sun::star::frame::XFrame> const&, chart::ChartController*&) [inlined] at /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS15.0.sdk/usr/include/c++/v1/memory:2099
#9	0x000000010068ae98 in chart::sidebar::ChartLinePanel::Create(weld::Widget*, com::sun::star::uno::Reference<com::sun::star::frame::XFrame> const&, chart::ChartController*) at /Users/tml/lo/co-2021-ios-optimised/chart2/source/controller/sidebar/ChartLinePanel.cxx:120
#10	0x000000010067c3b8 in chart::sidebar::ChartPanelFactory::createUIElement(rtl::OUString const&, com::sun::star::uno::Sequence<com::sun::star::beans::PropertyValue> const&) at /Users/tml/lo/co-2021-ios-optimised/chart2/source/controller/sidebar/Chart2PanelFactory.cxx:101
#11	0x00000001010e4128 in (anonymous namespace)::UIElementFactoryManager::createUIElement(rtl::OUString const&, com::sun::star::uno::Sequence<com::sun::star::beans::PropertyValue> const&) at /Users/tml/lo/co-2021-ios-optimised/framework/source/uifactory/uielementfactorymanager.cxx:437
#12	0x0000000102986e9c in sfx2::sidebar::SidebarController::CreateUIElement(com::sun::star::uno::Reference<com::sun::star::awt::XWindow> const&, rtl::OUString const&, bool, sfx2::sidebar::Context const&) at /Users/tml/lo/co-2021-ios-optimised/sfx2/source/sidebar/SidebarController.cxx:998
#13	0x0000000102986288 in sfx2::sidebar::SidebarController::CreatePanel(std::__1::basic_string_view<char16_t, std::__1::char_traits<char16_t> >, weld::Widget*, bool, sfx2::sidebar::Context const&, VclPtr<sfx2::sidebar::Deck> const&) at /Users/tml/lo/co-2021-ios-optimised/sfx2/source/sidebar/SidebarController.cxx:939
#14	0x0000000102985cf4 in sfx2::sidebar::SidebarController::CreatePanels(rtl::OUString const&, sfx2::sidebar::Context const&) at /Users/tml/lo/co-2021-ios-optimised/sfx2/source/sidebar/SidebarController.cxx:748
#15	0x0000000102985940 in sfx2::sidebar::SidebarController::CreateDeck(rtl::OUString const&, sfx2::sidebar::Context const&, bool) at /Users/tml/lo/co-2021-ios-optimised/sfx2/source/sidebar/SidebarController.cxx:700
#16	0x0000000102984e94 in sfx2::sidebar::SidebarController::SwitchToDeck(sfx2::sidebar::DeckDescriptor const&, sfx2::sidebar::Context const&) at /Users/tml/lo/co-2021-ios-optimised/sfx2/source/sidebar/SidebarController.cxx:853
#17	0x00000001029837d0 in sfx2::sidebar::SidebarController::UpdateConfigurations() at /Users/tml/lo/co-2021-ios-optimised/sfx2/source/sidebar/SidebarController.cxx:594
#18	0x00000001041733e8 in Link<void*, void>::Call(void*) const [inlined] at /Users/tml/lo/co-2021-ios-optimised/include/tools/link.hxx:111
#19	0x00000001041733e0 in ImplHandleUserEvent(ImplSVEvent*) [inlined] at /Users/tml/lo/co-2021-ios-optimised/vcl/source/window/winproc.cxx:2011
```

**Comment by tml1024** [comment]
Trying to build the app against core master to see if something has been changed there that prevents this crash, and in that case I will make sure the same fix is cherry-picked to the co-2021 branch, too. If no such obvious potential cherry-pick is found, I will just add a trivial check for a null pointer.

**Comment by tml1024** [comment]
Crashes the same way when built against core master, too.

**Comment by tml1024** [comment]
I suspect just adding `if (pLineStyleControl) ` etc checks here and there, bypassing code that tries to use the line style control, is not the right way to fix this. Sadly I don't really know much about how chart editing and how it is implemented so I will need to investigate and ask for advice about how it should be fixed.

**Comment by tml1024** [comment]
Fix in https://gerrit.libreoffice.org/c/core/+/122720

**Comment by tml1024** [comment]
The need to add new stuff to that native-code.py is a recurring problem when switching the iOS app to use a newer branch of core.

**Comment by nikslor** [comment]
Fixed in 21.6.1 (2) thanks a lot @tml1024. Do you want to close?
